apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -158,7 +158,9 @@ simple_async_query(Id, Request, QueryOpts0) ->
     Ref = make_request_ref(),
     Result = call_query(async_if_possible, Id, Index, Ref, ?SIMPLE_QUERY(Request), QueryOpts),
     _ = handle_query_result(Id, Result, _HasBeenSent = false),
-    Result.
+    maybe_apply_async_reply_fun(
```

Comment:
I think we should leave the calling of the reply function to `do_reply_caller`, and just inject the callback as the `From` here instead of `undefined`.

i.e., something like:

```erlang
  ReplyTo = maps:get(reply_to, QueryOpts, undefined),
  Query = ?QUERY(ReplyTo, Request, _HasBeenSeen = false, _ExpireAt = infinity),
  ...
```

and then I think `do_reply_caller` will call `{Fun, Args}` from `From` without further changes.